zkevm-circuits/src/evm_circuit.rs

```diff
@@ -1 +1,930 @@
 //! The EVM circuit implementation.
+
+use bus_mapping::operation::Target;
+use halo2::{
+    arithmetic::FieldExt,
+    circuit::{self, Layouter, Region},
+    plonk::{
+        Advice, Column, ConstraintSystem, Error, Expression, Fixed, Selector,
+    },
+    poly::Rotation,
+};
+use std::{convert::TryInto, iter};
+
+mod op_execution;
+use op_execution::{OpExecutionGadget, OpExecutionState};
+mod param;
+use param::{CIRCUIT_HEIGHT, CIRCUIT_WIDTH, NUM_CELL_OP_EXECTION_STATE};
+
+#[derive(Clone, Debug)]
+pub(crate) enum CallField {
+    // Transaction id.
+    TxId,
+    // Global counter at the begin of call.
+    GlobalCounterBegin,
+    // Global counter at the end of call, used for locating reverting section.
+    GlobalCounterEnd,
+    // Caller’s unique identifier.
+    CallerID,
+    // Offset of calldata.
+    CalldataOffset,
+    // Size of calldata.
+    CalldataSize,
+    // Depth of call, should be expected in range 0..=1024.
+    Depth,
+    // Address of caller.
+    CallerAddress,
+    // Address of code which interpreter is executing, could differ from
+    // `ReceiverAddress` when delegate call.
+    CodeAddress,
+    // Address of receiver.
+    ReceiverAddress,
+    // Gas given of call.
+    GasAvailable,
+    // Value in wei of call.
+    Value,
+    // If call succeeds or not in the future.
+    IsPersistant,
+    // If call is within a static call.
+    IsStatic,
+    // If call is `CREATE*`. We lookup op from calldata when is create,
+    // otherwise we lookup code under address.
+    IsCreate,
+}
+
+#[derive(Clone, Debug)]
+pub(crate) enum CallStateField {
+    // Program counter.
+    ProgramCounter,
+    // Stack pointer.
+    StackPointer,
+    // Memory size.
+    MemeorySize,
+    // Gas counter.
+    GasCounter,
+    // State trie write counter.
+    StateWriteCounter,
+    // Last callee's unique identifier.
+    CalleeID,
+    // Offset of returndata.
+    ReturndataOffset,
+    // Size of returndata.
+    ReturndataSize,
+}
+
+#[derive(Clone, Debug)]
+pub(crate) enum BusMappingLookup<F> {
```

Comment:
Seems that this could be externalized to the bus mapping as most of the stuff could be parsed there and you would not need to define it here.
WDYT?

Reply:
This is only for circuit constraint configuration, it's fixed in circuit. The things needed to be externalized to `bus-mapping` are `ExecutionStep` and `Operation` you mentioned below, they are actually parsed from `bus-mapping`

Reply:
My bad. Thanks for the explanation!